### DISCUSSION TITLE
IPv6...DHCP bug?

### DISCUSSION THREAD

**Discussion by samqixin** [fakemanhk/openwrt-jp-ipoe#17] [Q&A] [answered]
Thanks to fakemanhk for making instructions on IPOE in openwrt.
I was refering to other old version Japanese written guides for my own implementation.
I found there are 2 methods to add the ipv6-pd, one is adding to the config file as we have here, the other one is by adding a interface with static ipv6 address. Both methods do the job for map-e to run.

However, if I set wan6 and lan's ipv6 DHCP to all relay. PCs and iPhones in the LAN can access ipv6 all green for only a short period of time. Then everything fell to connect ipv6 anymore. 
I'm not sure it's related to ipv6-pd modification or current openwrt works bad with Japan NTT ipv6.

My workaround was using a buffalo router to first do the ipv6 "bridge", then use openwrt to do wan6, map-e and pppoe, which was working fine but I don't like the inpurity...
So I guess the blame goes to DHCPv6 and everything...?

**Comment by samqixin** [comment]
Pardon me if my long note confuses anyone.

I'm kinda blaming MWAN3 for making my ipv6 LAN clients not accessible, several hours after router boot.
If I restart MWAN3, let it reconfirm if all interface online, my PC can access ipv6 instantly.
Still trying.

**Comment by samqixin** [comment, marked as answer]
I have modify MWAN3 rules and it seems working fine now.

Dunno why but I followed old guides and added ipv6 rules in MWAN3, something like use the only wan6 for matching ::0/0.
If you have multiple WAN4 and have only 1 interface as WAN6, just ignore adding any ipv6 interface or rules in MWAN3.
My case is that I have a PPPoE WAN and a MAP-E WAN, so I only need to add ipv4 rules in MWAN3.
Other stuff like DHCPv6 plz follow other guides in the repo.

**Comment by fakemanhk** [comment]
> My workaround was using a buffalo router to first do the ipv6 "bridge", then use openwrt to do wan6, map-e and pppoe, which was working fine but I don't like the inpurity...

The Buffalo router can be IPv6 bridge? I didn't see such setting, can you point me where? But in this case you are adding one more hop to the path...
BTW did you check the other discussion that we can use SLAAC instead of DHCPv6? More client compatibility but your router won't see the client's IPv6 address from status page.

**Reply by samqixin** [reply]
Hi thanks for replying.

Yes bless Buffalo for that almost all newer routers offer IPv4overIPv6 function, aka v6plus and others. 
In my case I was using wxr-5950ax12 for its dual 10Gbe.
You should find Ipv6 bridge option in IPv6 tab in advance menu, along with NDP proxy option.
Yes there was one more hop for both MAP-E and PPPoE in my openwrt and I didn't like it.
That's why I kept searching for the answer, now I found it was my MWAN3 setting's fault.

Yes I am using SLAAC now instead of DHCPv6. 
Since I confirmed MWAN3 was to blame, I will try roll back to DHCPv6 to see if everything works:)

Btw I hated wxr-5950ax12 because it's so big and warm.
Now I switched to GL.iNet GL-MT3000, WiFi6 with 2.5Gbe, perfect as an AP in my network topology.
GL-MT3000 is also running openwrt so it might be able to provide me a compact all-in-one router, but x86 openwrt just rocks. lol

**Reply by fakemanhk** [reply]
The cheap Buffalo provided by my ISP doesn't have that IPv6  bridging  + NDP option unfortunately.

> Btw I hated wxr-5950ax12 because it's so big and warm.

Do you plan to sell it? lol.....I am interested in dual 10G, and compared with other dual 10G solution I have, the Buffalo should be already the lowest power consumption one.

Back to SLAAC vs DHCPv6, I don't think it would affect how your MWAN3 working since it's only about how a client getting it's IPv6 address and DNS information, assume you are setting fixed IPv6 on your client that should be the same (so did you ever tried fixed v6 address?)

BTW can you report your ISP info in Discussion #3 ? So that more people knows which one is known to work (though I believe most of them should work)

**Reply by samqixin** [reply]
Funny story, at first when I wanted to try v6plus I refered to a NEC router, which was the rental model by so-net.
But that didn't let me use port forwarding, then I went back to Buffalo, which kindly lists all usable ports lol.
Too bad I have other plans for the router.
Why don't you consider using seperate AP + 10G switch? The 2.5Gbe WiFi6 openwrt AP is perfect.
I also don't recommend the wxr-5950ax12 for its "dual" 10Gbe, 
they only let you use 1 for WAN and 1 for LAN in router mode,
and in AP mode the 2 ports don't even pass through packets.
I never successfully used the 2 10Gbe as switching ports ever.
Maybe it's just me, maybe you can try the updated WXR-6000AX12.

I agree that  SLAAC and DHCPv6 shouldn't affect MWAN3.
Since my case was the clients were getting IPv6 addresses and then fail to reach outside via IPv6,
maybe it's just MWAN3's ipv6 rules fault.

Yup I will report in the other discussion.

**Reply by fakemanhk** [reply]
> Why don't you consider using seperate AP + 10G switch? The 2.5Gbe WiFi6 openwrt AP is perfect.
> I also don't recommend the wxr-5950ax12 for its "dual" 10Gbe,
> they only let you use 1 for WAN and 1 for LAN in router mode,
> and in AP mode the 2 ports don't even pass through packets.
> I never successfully used the 2 10Gbe as switching ports ever.

I will be moving home soon and I can get 10G internet from the same provider for by just paying 600 YEN more per month.
Trying to look around and difficult to find any good and low power 10G solution (and I will be using it for 1 WAN + 1 LAN), in fact you probably didn't notice that the Buffalo WXR-5950AX12 which is IPQ8074A started to have OpenWrt [snapshot](https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=f490295bf213c5e181d16cbe53c163c1c9983e45) support since Jan 2023?? It has 1GB ram + 256MB flash which means it can handle a decent amount of connections without problem! I am planning to use it and run PiHole LXC on top to serve 10G internet.

In the meantime I am also asking some friends who are going to visit me later to bring me Mellanox Connect3 Dual 10G SFP+ card, and I am thinking about using some old Dell slim PC and swap an Intel T-series CPU for a complete 10G router solution but that would be much more power hungry when compared with the Buffalo router (I know if you need SQM then we still need x86 PC)

**Reply by samqixin** [reply]
They have support for the chip?!
Damn I lost faith in new Buffalo router ddwrt/openwrt support for a long time, Buffalo used to have official ddwrt support years ago.
Oh man thanks for the info! I'll definitely poke around.
happy for your 10G FTTH, I guess it’s NTT Flet’s?
I probably have to move to a house for 10G in the future lol.

My home network topology is basically a central all-in-one server with a X550-t2, pricy but I need the multi-gig support. It uses my old h170 MB and i5-6500t. I’m running my NAS as a VM so a x86 router costs nothing.

**Reply by samqixin** [reply]
I was also considering qnap QHora-301, which supports openwrt officially. It's also very expensive.

So my final build is from the server, 1 10Gbe to my 2.5Gbe PC and the other 10Gbe to the AP.

**Reply by fakemanhk** [reply]
> They have support for the chip?!

The support was removed previously due to some issue when I searched a bit, but in Jan 2023 the OpenWrt team was able to introduced this back to their master snapshot, so quite a number of Qualcomm IPQ807x were on list, including the mentioned QNAP QHora-301W, however I just found from Amazon for a used 5950AX12 (~22K Yen) so I would go for Buffalo (it has ext. antenna which should have better WiFi as well). The newer WXR-6000AX12 is reported to have same IPQ chipset but different bootloader so not on list yet.

**Reply by samqixin** [reply]
Nice.
Well I actually prefer internal antenna models for tidiness and I trust the signal since Netgear R6300v2.
That was quite a good deal consider that WXR-6000AX12 and QHora-301W cost so much more.
I'd love to see and discuss in future posts on your 5950AX12 openwrt implementation. 
I found an atempt on installing openwrt in 5950AX12 [here](https://tetsupc.wordpress.com/2023/02/25/buffalo-wxr-5950ax12/).
May try it later.

**Reply by fakemanhk** [reply]
> Well I actually prefer internal antenna models for tidiness and I trust the signal since Netgear R6300v2.

However this doesn't apply to QNAP (I know someone who owns this device and he's feeling upset about it's WiFi)
And if you don't care about WiFi, the iEi PUZZLE-M902 is the version without WiFi which is even better (and more expensive)

**Reply by samqixin** [reply]
Thanks.
I have just flashed my 5950AX12 using TFTP.
Everything seems working fine but I haven't tested 10Gbe throughput or Wifi yet

**Reply by fakemanhk** [reply]
> Thanks. I have just flashed my 5950AX12 using TFTP. Everything seems working fine but I haven't tested 10Gbe throughput or Wifi yet


I just received the delivery today, might try it out later after installing the new 10G broadband (I have to show them a working 10G one at installation to avoid paying 550Yen/month for the rental router)

**Reply by samqixin** [reply]
The rental is optional and you have to convince them not to rent it?
Aww who do they, the provider, think they are :(
Hope everything work out fine.

**Reply by fakemanhk** [reply]
> The rental is optional and you have to convince them not to rent it?

Technically they are not forcing me to rent, but on the phone they keep saying if I don't have a qualified one during installation then the installation process might be delayed blah blah blah.....which is annoying.

BTW I just found out that the 5950AX12 5GHz radio has 8x8 (80MHz) or 4x4 (160MHz) while the QHora-301W is 4x4 (80MHz) or 2x2 (160MHz), and because of ext. vs int. antenna, that makes the Buffalo one outperforms QNAP one. Of course if you plan to install many packages the QNAP one has more eMMC storage which would be great, but you can also put USB stick on Buffalo to make it as extended storage (I tested with other router with ext. USB stick to install PiHole to form a decent adblock solution)

**Reply by samqixin** [reply]
Hmm I didn't notice QNAP has only 4x4 and 2x2. (on the other hand that would be more than enough for home usage)
That would indeed make Buffalo the best one now with openwrt support.
I don't really need too much storage, can always put PiHole in VM or docker in NAS lol.

Will you post anything your 10G speed when it's done?

**Reply by fakemanhk** [reply]
> Will you post anything your 10G speed when it's done?

There were lots of delays and also because of Golden Week, my earliest date to have new 10G internet will be end of the month (and they told me the first couple of days only IPv6 will work and the IPv4 might need to wait another few days).

Just saw from OpenWrt forum, there is a possibility that the 10G port can't negotiate multi-gig (2.5G/5G), do you have this problem? Or you only have 10G so this is not an issue at all?

**Reply by samqixin** [reply]
I haven't touched it since the openwrt installation though...
Didn't see that coming, I did have multi-gig negotiation trouble with the Intel X550-T2 in my PVE machine, had to patch it manually.

**Reply by samqixin** [reply]
Heeeey, just a friendly reminder that 5950AX12 is now supported by openwrt in stable release 23.05.
https://downloads.openwrt.org/releases/23.05.0/targets/ipq807x/generic/

**Comment by fakemanhk** [comment]
> Heeeey, just a friendly reminder that 5950AX12 is now supported by openwrt in stable release 23.05. https://downloads.openwrt.org/releases/23.05.0/targets/ipq807x/generic/

Yes I knew about that, and I am looking some other people's custom build with NSS driver (the one with hardware acceleration)

**Reply by samqixin** [reply]
On the other hand QHora-301W is quite popular in China now, with great price, at about 25000yen. Some Chinese openwrt community's custom builds are supporting it too, some unlocks WiFi power and hardware acceleration. Too bad wxr-5950ax12 is replaced by new models.
